libs/networking/Http_mock_client.mli

```diff
@@ -79,3 +93,40 @@ val with_testing_client : make_response_fn -> (unit -> unit) -> unit -> unit
   * that uses [make_fn] to respond.
   * Any code inside [f] will be run with this modified client.
   *)
+
+val client_from_file :
+  string ->
+  (module Cohttp_lwt.S.Client)
+  * ((((Cohttp.Request.t * Cohttp_lwt.Body.t)
+      * (Cohttp.Response.t * Cohttp_lwt.Body.t))
+      list ->
+     'a) ->
+    'a)
+(** [client_from_file "request-response-file.network"]
+  * will return a new http client which may be used in lieu of the
+  * [http_helpers] client, which rather than performing network requests will
+  * instead look for a matching request in the given file and yield the
+  * given paired response. If no request matches, a test failure will be
+  * raised with Alcotest.
+  *
+  * The file format is
+  * ```
+  * > REQUEST (possibly multiple lines)
+  * < RESPONSE (possibly multiple lines)
```

Comment:
I'm ok with a custom file format here, but would appreciate more of an explanation of the format. It looks like this is just the actual bytes of the http request, but not sure, and I'd prefer not having to read parser code to figure it out.

Reply:
Yes, it is essentially the actual bytes modulo line endings. Do you not feel the later example clarifies sufficiently? My priority is actually being able to use this more so than documenting every function precisely.

Reply:
resolve so ci runs

Reply:
Can you add a comment about that? Just so people know how the test works in case they need to debug it